IBPSA/Fluid/HeatPumps/ModularReversible/Controls/Safety/BaseClasses/BoundaryMap.mo

```diff
@@ -0,0 +1,192 @@
+within IBPSA.Fluid.HeatPumps.ModularReversible.Controls.Safety.BaseClasses;
+model BoundaryMap
+  "Model which returns false if the input parameters
+  are out of the given charasteristic map"
+  parameter Real tab[:,2]
+    "Table for boundary with second column as useful temperature side";
+  parameter Real dT
+    "Delta value used to avoid state events when used as a safety control"
+  annotation (Dialog(tab="Safety Control", group="Operational Envelope"));
+  parameter Boolean isUppBou "=true if it is an upper boundary, false for lower";
+  Modelica.Blocks.Interfaces.BooleanOutput noErr
+    "If an error occurs, this will be false"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+  Modelica.Blocks.Interfaces.RealInput TNotUse(unit="K", displayUnit="degC")
+    "Not useful temperature side"
+    annotation (Placement(transformation(extent={{-130,-54},{-102,-26}})));
+
+  Modelica.Blocks.Tables.CombiTable1Ds tabBou(
+    final table=tab,
+    final smoothness=Modelica.Blocks.Types.Smoothness.LinearSegments,
+    final tableOnFile=false) "Table with envelope data"
+    annotation (Placement(transformation(extent={{-60,40},{-40,60}})));
+  Modelica.Blocks.MathBoolean.Nor nor(nu=3)
+    "If no condition violates the envelope, this is true"
+    annotation (Placement(transformation(extent={{60,-10},{80,10}})));
+  Modelica.Blocks.Logical.Hysteresis hysLef(
+    final uLow=-0.05,
+    final uHigh=0,
+    pre_y_start=false) "Hysteresis for left side of envelope"
+  annotation (Placement(transformation(extent={{20,-40},{40,-20}})));
+  Modelica.Blocks.Logical.Hysteresis hysRig(
+    final uLow=-0.05,
+    final uHigh=0,
+    pre_y_start=false) "Hysteresis for right side of envelope"
+    annotation (Placement(transformation(extent={{20,-80},{40,-60}})));
+  Modelica.Blocks.Sources.Constant conTNotUseMin(k=TNotUseMin)
+    "Constant minimal temperature of not useful temperature side"
```

Comment:
I could not find a good name either. I followed the `Carnot` approach in the IBPSA on this one, e.g. `TUseAct_nominal`. `NotUse` is mostly anergy, but this will only confuse users which do not know the concept of anergy and exergy.

Reply:
@FWuellhorst -
One more suggestion on this. What do you think of "load side" (replacing "useful side") & "source side" (replacing "not useful side")? The "source" has to be interpreted as source of heating or cooling, which can be marked in the description string.

Reply:
I thought about naming it source and sink, which is typical at least for heat pumps. 
However, when the operation is reversed, the source becomes the sink and vice versa. The same would apply for condenser&evaporator, load&source, etc.

However, the useful side will always stay at the same side (i.e. in the building). This is why I think this is the better naming approach.

Reply:
I will move this discussion to #1576